src/scene.rs

```diff
@@ -182,34 +181,27 @@ impl Scene {
         } = event {
             match keycode {
                 VirtualKeyCode::F => {
-                    self.camera.target = match self.camera.target {
-                        Target::Origin | Target::FabricMidpoint => {
-                            let [face_id] = fabric.faces
-                                .keys()
-                                .take(1)
-                                .next_chunk()
-                                .unwrap();
-                            Target::Face(*face_id)
+                    self.select_face(Some(match self.camera.target {
+                        Origin | FabricMidpoint | Hold => {
+                            *fabric.faces.keys().next().unwrap()
                         }
-                        Target::Face(face_id) => {
-                            let found = fabric.faces
-                                .keys()
+                        SelectedFace(face_id) => {
+                            let face_position = fabric.faces.keys()
                                 .position(|&id| face_id == id)
                                 .expect("Face id not found");
                             let &new_face_id = fabric.faces.keys()
                                 .cycle()
-                                .skip(found + 1)
-                                .next()
+                                .nth(face_position + 1)
                                 .unwrap();
-                            Target::Face(new_face_id)
+                            new_face_id
```

Comment:
You're relying on HashMap's keys having a stable order, [which isn't technically the case](https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.keys). Might be worth using an IndexMap instead.